R/exposure_pie_chart.R

```diff
@@ -0,0 +1,103 @@
+#' Create an interactive exposure pie chart in an htmlwidget
+#'
+#' @param .data a exposure pie data frame
+#' @param width width of exported htmlwidget in pixels (single integer value; default == NULL)
+#' @param height height of exported htmlwidget in pixels (single integer value; default == NULL)
+#' @param ... other options passed on to r2d3::r2d3() (see details)
+#'
+#' @description
+#' The `exposure_pie_chart` function creates an interactive exposure pie chart in an htmlwidget
+#'
+#' @md
+#' @export
+
+exposure_pie_chart <-
+  function(.data, width = NULL, height = NULL, ...) {
+    .data <- export_data_utf8(.data)
+
+    dependencies <-
+      list(
+        system.file("js/pie_exploded.js", package = "r2dii.interactive"),
+        system.file("js/d3-array.v2.min.js", package = "r2dii.interactive"),
+        system.file("js/d3-scale.v3.min.js", package = "r2dii.interactive"),
+        system.file("js/d3-scale-chromatic.v1.min.js", package = "r2dii.interactive"),
+        system.file("js/jquery-3.5.1.js", package = "r2dii.interactive"),
+        system.file("js/text_dropdown_jiggle.js", package = "r2dii.interactive"),
+        system.file("css/2dii_gitbook_style.css", package = "r2dii.interactive"),
+        system.file("css/hide_styles.css", package = "r2dii.interactive")
```

Comment:
Worth removing the duplication here? Maybe build a little helper `r2dii_interactive_path()` that is vectorized over the first argument so it works with something like this:

```r
r2dii_interactive_path(c(
 "js/pie_exploded.js",
 "js/d3-array.v2.min.js",
  ...
))
```

Reply:
I don't see a big benefit in that, and I prefer being able to see the actual command in line where it's in context rather than obscuring it with a custom function that is defined elsewhere.